pkg/armhelpers/deploymentError.go

```diff
@@ -7,12 +7,39 @@ import (
 	"strings"
 
 	"github.com/Azure/acs-engine/pkg/api"
-	"github.com/Azure/acs-engine/pkg/apierror"
 	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
 	"github.com/sirupsen/logrus"
 )
 
-func parseDeploymentOperation(logger *logrus.Entry, operation resources.DeploymentOperation) (*apierror.Error, error) {
+// ArmError is the OData v4 format, used by the RPC and
+// will go into the v2.2 Azure REST API guidelines
+type ArmError struct {
+	Code    string     `json:"code"`
+	Message string     `json:"message"`
+	Target  string     `json:"target,omitempty"`
```

Comment:
I don't think these and `Code`` are needed because `depErr.Error()` should have everything that's required. RP can deserialize it and ACS E can just return array to upgrade cmd that can then print it to stdout.

Reply:
No, this is not the case. ArmError is parsing ARM response. depErr.Error is a separate object